crates/transaction-pool/src/validate.rs

```diff
@@ -80,6 +77,137 @@ pub trait TransactionValidator: Send + Sync {
+impl<T: PoolTransaction + AccountProvider> TransactionValidator
+    for EthTransactionValidatorConfig<T>
+{
+    type Transaction = T;
+
+    async fn validate_transaction(
+        &self,
+        origin: TransactionOrigin,
+        transaction: Self::Transaction,
+    ) -> Result<TransactionValidationOutcome<Self::Transaction>, Error> {
+        let invalid_transaction = transaction.clone();
+
+        match transaction.tx_type() {
+            0 => {
+                // Accept legacy transactions
+            }
+
+            1 => {
+                // Accept only legacy transactions until EIP-2718/2930 activates
+                if !self.eip2718 {
+                    return Err(Error::TransactionEip2930Disabled)
+                }
+            }
+
+            2 => {
+                // Reject dynamic fee transactions until EIP-1559 activates.
+                if !self.eip1559 {
+                    return Err(Error::TransactionEip1559Disabled)
+                }
+            }
+
+            _ => return Err(Error::TxTypeNotSupported),
+        };
+
+        if self.shanghai {
+            match self.ensure_max_init_code_size(transaction.clone(), 2 * 24576) {
+                Ok(_) => {}
+                Err(e) => return Ok(TransactionValidationOutcome::Invalid(transaction, e)),
+            }
+        }
+
+        // Drop non-local transactions under our own minimal accepted gas price or tip
+        if !origin.is_local() && transaction.max_fee_per_gas() < self.gas_price {
+            return Err(Error::TransactionMaxFeeLessThenBaseFee)
+        }
+
+        // Ensure gasFeeCap is greater than or equal to gasTipCap.
+        if transaction.max_fee_per_gas() >= transaction.max_priority_fee_per_gas() {
+            return Err(Error::TransactionMaxFeeLessThenBaseFee)
+        }
+
+        // Checks for gas limit
+        if transaction.gas_limit() > self.current_max_gas_limit {
+            return Ok(TransactionValidationOutcome::Invalid(
+                transaction,
+                PoolError::TxExceedsGasLimit(
+                    *invalid_transaction.hash(),
+                    invalid_transaction.gas_limit(),
+                    self.current_max_gas_limit,
+                ),
+            ))
+        }
+
+        // transaction size
+        if transaction.size() > 4 * 32 * 1024 {
+            return Ok(TransactionValidationOutcome::Invalid(
+                transaction,
+                PoolError::TxExceedsMaxInitCodeSize(
+                    *invalid_transaction.hash(),
+                    invalid_transaction.size(),
+                    4 * 32 * 1024,
+                ),
+            ))
+        }
+
+        // Checks for chainid
+        if transaction.chain_id() != self.chain_id {
+            return Err(Error::TransactionChainId)
+        }
+
+        // Checks for sender
+        let account = match self.client.basic_account(transaction.sender())? {
+            Some(account) => account,
+            None => {
```

Comment:
this would happen if the account does not exist yet. how does geth handle this case?